File: 1161/Solution.cpp
#include <queue>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int maxSum = root->val;
        int maxLevel = 1;
        int currentLevel = 1;
        std::queue<TreeNode*> q;
        q.push(root);
        while (!q.empty())
        {
            int levelSize = q.size();
            int currentSum = 0;
            for (int i = 0; i < levelSize; ++i) {
                TreeNode* node = q.front();
                q.pop();
                currentSum += node->val;
                if (node->left) q.push(node->left);
                if (node->right) q.push(node->right);
            }
            if (currentSum > maxSum) {
                maxSum = currentSum;
                maxLevel = currentLevel;
            }
            ++currentLevel;
        }
        return maxLevel;
    }
};

File: 1161/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        maxSum = root.val
        maxLevel = 1
        currentLevel = 1
        queue = [root]
        while queue:
            currentSum = 0
            nextQueue = []
            for node in queue:
                currentSum += node.val
                if node.left:
                    nextQueue.append(node.left)
                if node.right:
                    nextQueue.append(node.right)
            if currentSum > maxSum:
                maxSum = currentSum
                maxLevel = currentLevel
            queue = nextQueue
            currentLevel += 1
        return maxLevel